src/functions/typedEnum.ts

```diff
@@ -0,0 +1,40 @@
+import { isArray } from 'lodash';
+import { IFunction, IFunctionResult, IRule, RuleFunction } from '../types';
+import { schema } from './schema';
+
+export type TypedEnumRule = IRule<RuleFunction.TYPED_ENUM>;
+
+export const typedEnum: IFunction = (targetVal, opts, paths, otherValues): void | IFunctionResult[] => {
+  const { enum: enumValues, ...initialSchema } = targetVal;
+
+  if (!isArray(enumValues)) {
```

Comment:
@philsturgeon We currenlty bail out if not an array. @P0lip mentioned that we may also report an malformed enum. However, shouldn't a malformed enum be identified by the `oas[2|3]-schema` rule?

Reply:
@nulltoken yeah I think this rule has a very specific target of "make sure the array items match the type defined" so other stuff should be another.